protocol/x/clob/keeper/liquidations.go

```diff
@@ -105,7 +106,18 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 			}
 
 			// Return unexpected errors.
-			return err
+			return 0, err
+		}
+
+		// Skip liquidation if the ClobPair for the liquidation order is in final settlement. There is a separate flow
+		// for triggering deleveraging/final settlement for positions in closed markets.
+		if _, found := clobPairStatuses[liquidationOrder.GetClobPairId()]; !found {
+			clobPair := k.mustGetClobPair(ctx, liquidationOrder.GetClobPairId())
+			clobPairStatuses[liquidationOrder.GetClobPairId()] = clobPair.Status
+		}
+		if clobPairStatuses[liquidationOrder.GetClobPairId()] == types.ClobPair_STATUS_FINAL_SETTLEMENT {
+			unfilledLiquidations = append(unfilledLiquidations, *liquidationOrder)
+			continue
```

Comment:
nit: For normal orders, trading is disabled within the `keeper.PlaceOrder` flow(they would fail clob pair status transition validation). for consistency, this logic should also live inside `keeper.PlacePerpetualLiquidation`

Reply:
`PerformStatefulOrderValidation` is where we call `validateOrderAgainstClobPairStatus`, the validation you are referring to. We don't currently call stateful validation when placing liquidation orders, instead we skip straight to calling `matchOrder` because the liquidation order is protocol generated and therefore cannot / does not need to be validated in the same way a regular order does.

The `validateOrderAgainstClobPairStatus` fn expects an `Order` which cannot be fetched from a liquidation order. 

I feel like keeping this as is would be simpler. If I have the `PlacePerpetualLiquidation` fn return an error depending on clob pair status then we also need to properly recover from this error  in `LiquidateSubaccountsAgainstOrderbook` when we could have just skipped the function call in the first place

Reply:
Putting it in `PlacePerpetualLiquidation` will make it easier to write a unit test, e.g. if we decide to write a regression test or something. Error handling should be trivial.

but yeah idk. `LiquidateSubaccountsAgainstOrderbook` is a top level function, so I feel like this implementation detail should live in a downstream function, such as `keeper.PlacePerpetualLiquidation`. If you want this to fail earlier or get skipped more upstream, this logic can also be inside `GetPerpetualPositionToLiquidate` (additional benefit is that we would be picking another perpetual position to liquidate like Lucas mentioned).

Will let you make the call here.

Reply:
I can push it downstream. I don't want it upstream because I actually do want the liquidations flow to pick up negative TNC subaccounts in markets that are in final settlement and deleverage them normally (at bankruptcy price). So would prefer to not skip these and find a different position